evalml/data_checks/outliers_data_check.py

```diff
@@ -50,20 +52,136 @@ def validate(self, X, y=None):
+        # gamma distribution given the number of records in the data.
+        # For both measures, the values are are from a least squares regression
+        # model
+        log_n = np.log(num_records)
+        log_shape = (
+            25.8218734380722 +
+            -29.2320460088643 * log_n +
+            14.8228030299864 * log_n ** 2 +
+            -4.08052512660036 * log_n ** 3 +
+            0.641429075842177 * log_n ** 4 +
+            -0.0571252717322226 * log_n ** 5 +
+            0.00268694343911156 * log_n ** 6 +
+            -5.19415149920567e-05 * log_n ** 7
+        )
+        shape_param = np.exp(log_shape)
+        log_scale = (
+            -19.8196822259052 +
+            8.5359212447622 * log_n +
+            -8.80487628113388 * log_n ** 2 +
+            2.27711870991327 * log_n ** 3 +
+            -0.344443407676357 * log_n ** 4 +
+            0.029820831994345 * log_n ** 5 +
+            -0.00136611527293756 * log_n ** 6 +
+            2.56727158170901e-05 * log_n ** 7
+        )
+        scale_param = np.exp(log_scale)
+
+        # calculate and return the probability of no true outliers for a gamma
+        # cumulative density function
+        prob_val = 1.0 - gamma.cdf(pct_outliers, shape_param, scale=scale_param)
+        return prob_val
+
+    @staticmethod
+    def _outlier_score(column, convert_column=False):
+        """Return a dictionary of high and low values of potential numeric outliers using the IQR method.
+
+        Original credit goes to Jad Raad and Dan Putler of Alteryx.
+
+        Arguments:
+            column (pd.Series): A column of data to check for outliers in
+            convert_column (bool): If True, convert column to np.int64, if possible.
+
+        Returns:
+            dict: Dictionary containing outlier information
+        """
+        column_nonan = column[~pd.isna(column)]
+
+        if column_nonan.shape[0] == 0:
+            return None
+        else:
+            if convert_column:
+                column_nonan = column_nonan.astype(np.int64)
+
+            q1, median, q3 = np.percentile(column_nonan, [25, 50, 75])
+            column_iqr = q3 - q1
+
+            low_bound = q1 - (column_iqr * 1.5)
+            high_bound = q3 + (column_iqr * 1.5)
+
+            low_filter = column_nonan < low_bound
+            high_filter = column_nonan > high_bound
+
+            low_indices = column_nonan[low_filter].index.tolist()
+            high_indices = column_nonan[high_filter].index.tolist()
+
+            low_values = column.filter(low_indices).tolist()
+            high_values = column.filter(high_indices).tolist()
+
+            # calculate outlier probability
+            pct_outliers = (len(low_values) + len(high_values)) / len(column_nonan)
+
+            num_records = len(column_nonan)
+            score = OutliersDataCheck._no_outlier_prob(num_records, pct_outliers)
+            result = {
+                "score": score,
+                "values": {
```

Comment:
I think we should get rid of most of the stuff in `values` since the `validate` method only checks the score. 

That being said, returning at least a sample of `low_indices` and `high_indices` could be useful for #1730 and users to decide what to do about the outliers.

Reply:
RE comment above, this is mostly to align with what's being used elsewhere, so the same functionality is available in EvalML when we converge :)